demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* Standard includes. */
+#include <stdio.h>
+#include <stdint.h>
+
+/* POSIX socket includes. */
+#include <netdb.h>
+#include <unistd.h>
+
+#include <sys/socket.h>
+#include <sys/types.h>
+
+#include "config.h"
+
+#include "http_client.h"
+
+/**
+ * @brief HTTP server host name.
+ *
+ * This demo uses httpbin.org: A simple HTTP Request & Response Service.
+ */
+#define SERVER                      "httpbin.org"
+
+/**
+ * @brief HTTP server port number.
+ *
+ * In general, port 80 is for plaintext HTTP connections.
+ */
+#define PORT                        80
+
+/**
+ * @brief Paths for different HTTP methods for specified host.
+ *
+ * For httpbin.org, see http://httpbin.org/#/HTTP_Methods for details on
+ * supported REST API.
+ */
+#define GET_PATH                    "/ip"
```

Comment:
Since we have `#ifdef` for these paths in main(), I think these will need `#ifndef..#endf` around them here. 
If these are not planned to be configurable we need to take out the `#ifdef` constructs from main().